GramAddict/plugins/data_analytics.py

```diff
@@ -0,0 +1,215 @@
+import json
+import matplotlib.pyplot as plt
+import os
+
+from datetime import timedelta, datetime
+from enum import Enum, unique
+from matplotlib import ticker
+from matplotlib.backends.backend_pdf import PdfPages
+from matplotlib.dates import DateFormatter
+
+from GramAddict.core.plugin_loader import Plugin
+
+A4_WIDTH_INCHES = 8.27
+A4_HEIGHT_INCHES = 11.69
+
+
+class DataAnalytics(Plugin):
+    """Generates a PDF analytics report of specified username session data"""
+
+    def __init__(self):
+        super().__init__()
+        self.description = (
+            "Generates a PDF analytics report of specified username session data"
+        )
+        self.arguments = [
+            {
+                "arg": "--analytics",
+                "nargs": 1,
+                "help": "generates a PDF analytics report of specified username session data",
+                "metavar": "username1",
+                "default": None,
+                "operation": True,
+            },
+        ]
+
+    def run(self, device, device_id, args, enabled, storage, sessions):
+        self.username = args.analytics[0]
+        sessions = self.load_sessions()
+        if not sessions:
+            return
+
+        filename = (
+            "report_"
+            + self.username
+            + "_"
+            + datetime.now().strftime("%Y-%m-%d")
+            + ".pdf"
+        )
+        with PdfPages(filename) as pdf:
+            sessions_week = self.filter_sessions(sessions, Period.LAST_WEEK)
+            sessions_month = self.filter_sessions(sessions, Period.LAST_MONTH)
+
+            self.plot_followers_growth(
+                sessions_week, pdf, self.username, Period.LAST_WEEK
+            )
+            self.plot_followers_growth(
+                sessions_month, pdf, self.username, Period.LAST_MONTH
+            )
+            self.plot_followers_growth(sessions, pdf, self.username, Period.ALL_TIME)
+
+            self.plot_duration_statistics(
+                sessions_week, pdf, self.username, Period.LAST_WEEK
+            )
+            self.plot_duration_statistics(
+                sessions_month, pdf, self.username, Period.LAST_MONTH
+            )
+            self.plot_duration_statistics(sessions, pdf, self.username, Period.ALL_TIME)
+
+        print("Report saved as " + filename)
+
+    def load_sessions(self):
+        path = self.username + "/sessions.json"
```

Comment:
We should provide an API for fetching the path for the session. The way it is right now makes it difficult to refactor.
Perhaps we can improve this. This kind of information should be fetched from the core.